deepspeed/runtime/engine.py

```diff
@@ -122,12 +122,12 @@ def split_half_float_double_sparse(tensors):
     supported_types = get_accelerator().supported_dtypes()
 
     for t in tensors:
-        assert t.dtype in supported_types, f"attempting to reduce an unsupported grad type: {t.dtype}"
+        assert t.dtype in supported_types if not isinstance(t, str), f"attempting to reduce an unsupported grad type: {t.dtype}"
```

Comment:
@abhilash1910, do you know why a t of type str showed up here?

Reply:
yes quite strange, where this is coming from as the supported_dtypes (cuda/cpu) are not in string